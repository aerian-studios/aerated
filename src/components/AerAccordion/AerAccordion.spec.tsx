import React from "react";
import { describe, it, expect, afterEach } from "vitest";
import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { AerAccordion } from "./index";

describe("AerAccordion", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render correctly", () => {
    const { container } = render(
      <AerAccordion.Root type="multiple">
        <AerAccordion.Item value="item-1">
          <AerAccordion.Header>Header 1</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
        <AerAccordion.Item value="item-2">
          <AerAccordion.Header>Header 2</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
        <AerAccordion.Item value="item-3">
          <AerAccordion.Header>Header 3</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
      </AerAccordion.Root>
    );
    expect(container).toMatchSnapshot();

    expect(
      screen.getByRole("heading", { name: "Header 1", level: 3 })
    ).toBeDefined();
    expect(
      screen.getByRole("heading", { name: "Header 2", level: 3 })
    ).toBeDefined();
    expect(
      screen.getByRole("heading", { name: "Header 3", level: 3 })
    ).toBeDefined();
  });

  it("should render the multiple variant by default", async () => {
    const { container } = render(
      <AerAccordion.Root>
        <AerAccordion.Item value="item-1">
          <AerAccordion.Header>Header 1</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
        <AerAccordion.Item value="item-2">
          <AerAccordion.Header>Header 2</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
        <AerAccordion.Item value="item-3">
          <AerAccordion.Header>Header 3</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
      </AerAccordion.Root>
    );
    expect(container).toMatchSnapshot();

    const headder1Trigger = screen.getByRole("button", { name: "Header 1" });
    const headder2Trigger = screen.getByRole("button", { name: "Header 2" });

    expect(headder1Trigger.getAttribute("aria-expanded")).toBe("false");
    expect(headder2Trigger.getAttribute("aria-expanded")).toBe("false");

    await userEvent.click(headder1Trigger);

    expect(headder1Trigger.getAttribute("aria-expanded")).toBe("true");
    expect(headder2Trigger.getAttribute("aria-expanded")).toBe("false");

    await userEvent.click(headder2Trigger);

    expect(headder1Trigger.getAttribute("aria-expanded")).toBe("true");
    expect(headder2Trigger.getAttribute("aria-expanded")).toBe("true");

    await userEvent.click(headder2Trigger);

    expect(headder1Trigger.getAttribute("aria-expanded")).toBe("true");
    expect(headder2Trigger.getAttribute("aria-expanded")).toBe("false");
  });

  it("should render the single variant when passed as a prop", async () => {
    const { container } = render(
      <AerAccordion.Root type="single">
        <AerAccordion.Item value="item-1">
          <AerAccordion.Header>Header 1</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
        <AerAccordion.Item value="item-2">
          <AerAccordion.Header>Header 2</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
        <AerAccordion.Item value="item-3">
          <AerAccordion.Header>Header 3</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
      </AerAccordion.Root>
    );
    expect(container).toMatchSnapshot();

    const headder1Trigger = screen.getByRole("button", { name: "Header 1" });
    const headder2Trigger = screen.getByRole("button", { name: "Header 2" });

    expect(headder1Trigger.getAttribute("aria-expanded")).toBe("false");
    expect(headder2Trigger.getAttribute("aria-expanded")).toBe("false");

    await userEvent.click(headder1Trigger);

    expect(headder1Trigger.getAttribute("aria-expanded")).toBe("true");
    expect(headder2Trigger.getAttribute("aria-expanded")).toBe("false");

    await userEvent.click(headder2Trigger);

    expect(headder1Trigger.getAttribute("aria-expanded")).toBe("false");
    expect(headder2Trigger.getAttribute("aria-expanded")).toBe("true");

    await userEvent.click(headder1Trigger);

    expect(headder1Trigger.getAttribute("aria-expanded")).toBe("true");
    expect(headder2Trigger.getAttribute("aria-expanded")).toBe("false");
  });

  it("should render heading overrides correctly", () => {
    const { container } = render(
      <AerAccordion.Root type="multiple">
        <AerAccordion.Item value="item-1">
          <AerAccordion.Header headingLevel="h2">Header 1</AerAccordion.Header>
          <AerAccordion.Content>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          </AerAccordion.Content>
        </AerAccordion.Item>
      </AerAccordion.Root>
    );

    expect(
      screen.queryByRole("heading", { name: "Header 1", level: 3 })
    ).toBeNull();
    expect(
      screen.getByRole("heading", { name: "Header 1", level: 2 })
    ).toBeDefined();
  });
});
