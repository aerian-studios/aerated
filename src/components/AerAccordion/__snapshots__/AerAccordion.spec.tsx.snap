// Vitest Snapshot v1

exports[`AerAccordion > should render correctly 1`] = `
<div>
  <div
    class="_accordionRoot_d92853"
  >
    <div
      data-state="closed"
    >
      <h3
        class="_accordionHeader_d92853"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r1:"
          aria-expanded="false"
          class="_accordionTrigger_d92853"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          Header 1
          <svg
            aria-hidden="true"
            class="_accordionChevron_d92853"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r0:"
        class="_accordionContent_d92853"
        data-state="closed"
        hidden=""
        id="radix-:r1:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      data-state="closed"
    >
      <h3
        class="_accordionHeader_d92853"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="false"
          class="_accordionTrigger_d92853"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r2:"
          type="button"
        >
          Header 2
          <svg
            aria-hidden="true"
            class="_accordionChevron_d92853"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r2:"
        class="_accordionContent_d92853"
        data-state="closed"
        hidden=""
        id="radix-:r3:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      data-state="closed"
    >
      <h3
        class="_accordionHeader_d92853"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r5:"
          aria-expanded="false"
          class="_accordionTrigger_d92853"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r4:"
          type="button"
        >
          Header 3
          <svg
            aria-hidden="true"
            class="_accordionChevron_d92853"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r4:"
        class="_accordionContent_d92853"
        data-state="closed"
        hidden=""
        id="radix-:r5:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</div>
`;
